SQF-VM/SQF_commands.c

```diff
@@ -3000,79 +2801,376 @@ void CMD_SET(void* input, CPCMD self)
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
 		return;
 	}
 	arrl = left_val->val.ptr;
 	arrr = right_val->val.ptr;
-	if (arrr->top < 1 || arrr->data[0]->type != SCALAR_TYPE())
-	{
-		vm->error(vm, ERR_ERR ERR_ARRAY_(0) ERR_WAS_EXPECTED ERR_SCALAR, vm->stack);
+	if (arrr->top < 1 || arrr->data[0]->type != SCALAR_TYPE()) {
+		vm->error(vm, ERR_ERR ERR_ARRAY_(0) ERR_WAS_EXPECTED ERR_SCALAR,
+				vm->stack);
 		inst_destroy(left);
 		inst_destroy(right);
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
 		return;
 	}
-	if (arrr->top < 2)
-	{
-		vm->error(vm, ERR_ERR ERR_ARRAY_(1) ERR_WAS_EXPECTED ERR_EXISTING, vm->stack);
+	if (arrr->top < 2) {
+		vm->error(vm, ERR_ERR ERR_ARRAY_(1) ERR_WAS_EXPECTED ERR_EXISTING,
+				vm->stack);
 		inst_destroy(left);
 		inst_destroy(right);
-		push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
 		return;
 	}
 	index = floor(arrr->data[0]->val.f);
 	val = arrr->data[1];
-	if (index >= arrl->size)
-	{
+	if (index >= arrl->size) {
 		array_resize(arrl, index + 1);
-		for (i = arrl->top; i <= index; i++)
-		{
+		for (i = arrl->top; i <= index; i++) {
 			arrl->data[i] = 0;
 		}
 	}
-	for (i = arrl->top; i < index; i++)
-	{
+	for (i = arrl->top; i < index; i++) {
 		array_push(arrl, value(NOTHING_TYPE(), base_int(0)));
 	}
-	if (arrl->data[index] != 0)
-	{
+	if (arrl->data[index] != 0) {
 		inst_destroy_value(arrl->data[index]);
 	}
-	if (arrl->top == index)
-	{
+	if (arrl->top == index) {
 		arrl->top++;
 	}
 	arrl->data[index] = value_copy(val);
-	
 
 	push_stack(vm, vm->stack, inst_value(value(NOTHING_TYPE(), base_int(0))));
 	inst_destroy(left);
 	inst_destroy(right);
 }
 
+void CMD_RESIZE(void* input, CPCMD self) {
+	PVM vm = input;
+	PINST left;
+	PINST right;
+	PVALUE left_val;
+	PVALUE right_val;
+
+	left = pop_stack(vm, vm->work);
+	right = pop_stack(vm, vm->work);
+	left_val = get_value(vm, vm->stack, left);
+	right_val = get_value(vm, vm->stack, right);
+
+	if (left_val == 0 || right_val == 0) {
+		inst_destroy(left);
+		inst_destroy(right);
+		return;
+	}
+
+	if (left_val->type != ARRAY_TYPE()) {
+		vm->error(vm, ERR_LEFT_TYPE ERR_ARRAY, vm->stack);
+
+		inst_destroy(left);
+		inst_destroy(right);
 
-void CMD_GETVARIABLE(void* input, CPCMD self)
-{
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+		return;
+	}
+
+	if (right_val->type != SCALAR_TYPE()) {
+		vm->error(vm, ERR_RIGHT_TYPE ERR_SCALAR, vm->stack);
+		push_stack(vm, vm->stack,
+				inst_value(value(NOTHING_TYPE(), base_int(0))));
+
+		inst_destroy(left);
+		inst_destroy(right);
+
+		return;
+	}
+
+	PARRAY array = left_val->val.ptr;
+	int newSize = right_val->val.f;
```

Comment:
Please declare all used variables in the functions begining before first non-declaring line